### ISSUE TITLE
update the coding standard to require docstrings for all new unit test methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1882]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1882|
|Type|task|
|Created|2006-06-30 13:43:15Z|
|Branch|https://github.com/twisted/twisted/tree/tests-docstring-standard-1882|

All unit test methods should have docstrings specifying at a high level the intent of the test. That is, a description that users of the method would understand.

<details><summary>Searchable metadata</summary>

```
trac-id__1882 1882
type__task task
reporter__radix radix
priority__high high
milestone__ 
branch__branches_tests_docstring_standard_1882 branches/tests-docstring-standard-1882
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1151674995000000 1151674995000000
changetime__1218997716000000 1218997716000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

Prerequisite: make trial not output the docstring instead of the test name?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Both the [doc/development/policy/test-standard.xhtml test](doc/development/policy/test-standard.xhtml test) and [doc/development/policy/coding-standard.xhtml coding](doc/development/policy/coding-standard.xhtml coding) standard docs need an update.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [24557]) Branching to 'tests-docstring-standard-1882'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_6"></a>|@washort set owner to @thijstriemstra|
|-|-|

I guess that works. merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24613]) Merge tests-docstring-standard-1882: updated the coding standard to require docstrings for all new unit test methods.

Author: thijs
Reviewer: washort
Fixes: [#1882](https://github.com/twisted/twisted/issues/1882)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
